src/teacher.py

```diff
@@ -72,3 +74,32 @@ def _answer_Demonstration_query(
                                           _is_demo=True,
                                           _train_task_idx=query.train_task_idx)
         return DemonstrationResponse(query, teacher_traj)
+
+
+@dataclass
+class TeacherInteractionMonitor(Monitor):
+    """Wraps the interaction between agent and teacher."""
```

Comment:
expand this docstring, e.g.
"""Wraps the interaction between agent and teacher to include generating and answering queries."""